spec/lib/additional_dependents_pdf_spec.rb

```diff
@@ -0,0 +1,150 @@
+          xml.EligibleForChildTaxCreditInd "X"
+        end
+        xml.DependentDetail do
+          xml.DependentFirstNm "Yet Another not on Return"
+          xml.DependentLastNm "Dependent"
+          xml.DependentNameControlTxt "Not on Return Dependent"
+          xml.DependentSSN "111111113"
+          xml.DependentRelationshipCd "DAUGHTER"
+          xml.EligibleForChildTaxCreditInd "X"
+        end
+        xml.DependentDetail do
+          xml.DependentFirstNm "Last one not on Return"
+          xml.DependentLastNm "Dependent"
+          xml.DependentNameControlTxt "Not on Return Dependent"
+          xml.DependentSSN "111111114"
+          xml.DependentRelationshipCd "DAUGHTER"
+          xml.EligibleForChildTaxCreditInd "X"
+        end
+        xml.DependentDetail do
+          xml.DependentFirstNm "FirstAdditional"
+          xml.DependentLastNm "Dependent"
+          xml.DependentNameControlTxt "FirstAdditional Dependent"
+          xml.DependentSSN "111111115"
+          xml.DependentRelationshipCd "SON"
+          xml.EligibleForChildTaxCreditInd "X"
+        end
+        xml.DependentDetail do
+          xml.DependentFirstNm "SecondAdditional"
+          xml.DependentLastNm "Dependent"
+          xml.DependentNameControlTxt "SECONDAdditional Dependent"
+          xml.DependentSSN "111111116"
+          xml.DependentRelationshipCd "FOSTER CHILD"
+          xml.EligibleForChildTaxCreditInd ""
+        end
+        xml.DependentDetail do
+          xml.DependentFirstNm "ThirdAdditional"
+          xml.DependentLastNm "Dependent"
+          xml.DependentNameControlTxt "ThirdAdditional Dependent"
+          xml.DependentSSN "111111117"
+          xml.DependentRelationshipCd "GRANDCHILD"
+          xml.EligibleForChildTaxCreditInd ""
+        end
+        xml.DependentDetail do
+          xml.DependentFirstNm "FourthAdditional"
+          xml.DependentLastNm "Dependent"
+          xml.DependentNameControlTxt "FourthAdditional Dependent"
+          xml.DependentSSN "111111118"
+          xml.DependentRelationshipCd "PARENT"
+          xml.EligibleForChildTaxCreditInd ""
+        end
+      end
+    end.doc
+  end
+
+  before do
+    allow_any_instance_of(SubmissionBuilder::Ty2021::Documents::Irs1040).to receive(:document).and_return fake_xml_document
+  end
+
+  describe "#output_file" do
+    context "without values" do
+      let(:fake_xml_document) do
+        Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
+          xml.RootNode do
+          end
+        end.doc
+      end
+
+      it "returns default values" do
+        output_file = pdf.output_file
+        empty = {}
+        22.times do |i|
+          empty.merge!({
+            "DependentNameRow#{i + 1}" => nil,
+            "CTCRow#{i + 1}" => nil,
+            "OTCRow#{i + 1}" => nil,
+            "RelationshipRow#{i + 1}" => nil,
+            "TINRow#{i + 1}" => nil,
+          })
+        end
+        expect(non_preparer_fields(output_file.path)).to eq empty
```

Comment:
It's probably better to use `filled_in_values(filepath).keys` rather than `non_preparer_fields`. I think the word `preparer` in `non_preparer_fields` only has meaning in the 13614-C. Not a blocker.